stream.go

```diff
@@ -599,6 +599,19 @@ func (ss *serverStream) SendMsg(m interface{}) (err error) {
 			}
 			ss.mu.Unlock()
 		}
+		if err != nil && err != io.EOF {
+			st, ok := status.FromError(err)
+			if !ok {
```

Comment:
Should we use toRPCErr for this conversion instead?  It does the same things but more, except for calling `convertCode`.  But that function says it's intended for use on errors returned by the server application, which is not the case here.

So something like this?

```go
st, _ := status.FromError(toRPCErr(err))
ss.t.WriteStatus(ss.s, st)
```